src/XrdHttp/XrdHttpUtils.hh

```diff
@@ -79,15 +83,97 @@ bool Fromhexdigest(const unsigned char *input, int length, unsigned char *out);
 
 void Tobase64(const unsigned char *input, int length, char *out);
 
-
 // Create a new quoted string
 char *quote(const char *str);
-
 // unquote a string and return a new one
 char *unquote(char *str);
 
-
-
+/**
+ * Creates a non-const copy of the string passed in parameter and
+ * calls unquote() on it before returning the pointer
+ * to the unquoted string
+ * @param str the string to unquote
+ * @return the malloc'd and unquoted string
+ * !!! IT MUST BE FREED AFTER USAGE USING free(...) !!!
+ */
+inline char * decode_raw(const char * str) {
+  if(!str) {
+    return nullptr;
+  }
+  size_t strLength = strlen(str);
+  // uniquely own the temporary copy
+  std::unique_ptr<char[]> buf(new char[strLength + 1]);
+  std::memcpy(buf.get(), str, strLength + 1);
```

Comment:
I wonder why you preferred not to use a `std::string` here.

Reply:
It's because I cannot (look at the `unquote()` function, it does modify the char * passed in parameter).